IM_calculation/IM/im_calculation.py

```diff
@@ -110,6 +110,28 @@ def calculate_rotd(
     return value_dict
 
 
+def compute_adv_measure(waveform, advanced_im_config, output_dir):
+    """
+    Wrapper function to call advanced IM workflow
+    :param waveform: Tuple of waveform objects (first is acc, second is vel)
+    :param advanced_im_config: advanced_im_config Named Tuple containing list of IMs, config file and path to OpenSeeS
+    :param output_dir: Directory where output folders are contained. Structure is /path/to/output_dir/station/im_name
+    :return:
+    """
+
+    if (advanced_im_config is not None) and (advanced_im_config.IM_list is not None):
+        waveform_acc = waveform[0]
+        station_name = waveform_acc.station_name
+        adv_im_out_dir = os.path.join(output_dir, station_name)
+        advanced_IM_factory.compute_ims(
+            waveform_acc, advanced_im_config, adv_im_out_dir
+        )
+    else:
+        raise AttributeError(
+            f"cannot access advanced_im_config doesnt contains IM_list {advanced_im_config}"
+        )
```

Comment:
Could use de morgans law to swap the logic, so you're testing for failure and raising, then can deindent the actual body?
Non functional change, just aesthetic

Reply:
Agreed with this change. The less indented blocks the better. Especially for terminating conditions.

Reply:
Does this function even need integrity checks? It should raise exceptions if the data isn't good., it's not like we're putting a default value in.
I think the onus should be on calling functions to ensure they're putting in good information